src/OpenSage.Game/Gui/Apt/ActionScript/Opcodes/Object.cs

```diff
@@ -100,18 +117,44 @@ public override void Execute(ActionContext context)
         }
     }
 
+    /// <summary>
+    /// get a property and push it to the stack
+    /// </summary>
     public sealed class GetProperty : InstructionBase
     {
         public override InstructionType Type => InstructionType.GetProperty;
 
         public override void Execute(ActionContext context)
         {
-            throw new NotImplementedException();
+            var property = context.Stack.Pop();
+            var target = context.GetTarget(context.Stack.Pop().ToString());
+            var type = (PropertyType) property.ToInteger();
```

Comment:
If this is a common thing to do, would it be worth having a `property.ToEnum<TEnum>()` method?

Reply:
property is of type Value, where i could add that. But the Value type has lots of room for improvements anyways ;D